### PR TITLE
Sage-intacct Payments: Auto Sync

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

**Review by Sravanksk — APPROVED** [review]

File: apps/fyle/migrations/0006_merge_20210204_1933.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('fyle', '0005_auto_20210201_1527'),
+        ('fyle', '0005_expense_billable'),
+    ]
+
+    operations = [
+    ]
```
Comment (ashwin1111) [inline apps/fyle/migrations/0006_merge_20210204_1933.py:14]: why is this empty?

File: apps/fyle/serializers.py [hunk 2]
```diff
@@ -35,6 +38,17 @@ class ExpenseFieldSerializer(serializers.ModelSerializer):
     """
     Expense Fields Serializer
     """
+
     class Meta:
         model = ExpenseAttribute
         fields = ['attribute_type', 'display_name']
+
+
+class ReimbursementSerializer(serializers.ModelSerializer):
```
Comment (ashwin1111) [inline apps/fyle/serializers.py:47]: are we using this anywhere? if not, please to remove. I think this is not exposed on app side.

File: apps/mappings/migrations/0004_merge_20210204_1933.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('mappings', '0003_auto_20210201_1604'),
+        ('mappings', '0003_auto_20210127_1551'),
+    ]
+
+    operations = [
```
Comment (ashwin1111) [inline apps/mappings/migrations/0004_merge_20210204_1933.py:13]: same, why

File: apps/mappings/utils.py [hunk 4]
```diff
@@ -18,14 +20,14 @@ def create_or_update_general_mapping(self, general_mapping: Dict):
...
-            general_settings.corporate_credit_card_expenses_object == 'CHARGE_CARD_TRANSACTION':
+                general_settings.corporate_credit_card_expenses_object == 'CHARGE_CARD_TRANSACTION':
             assert_valid('default_charge_card_name' in general_mapping and general_mapping['default_charge_card_name'],
                          'default charge card name field is blank')
             assert_valid('default_charge_card_id' in general_mapping and general_mapping['default_charge_card_id'],
                          'default charge card id field is blank')
 
-        elif general_settings.corporate_credit_card_expenses_object and \
-            general_settings.corporate_credit_card_expenses_object == 'BILL':
+        if general_settings.corporate_credit_card_expenses_object and \
```
Comment (ashwin1111) [inline apps/mappings/utils.py:29]: this can remain as elif, why is this changed to if?
corporate_credit_card_expenses_object can be either be BILL / CHARGE_CARD_TRANSACTION

File: apps/sage_intacct/migrations/0005_merge_20210204_1933.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('sage_intacct', '0004_auto_20210127_1345'),
+        ('sage_intacct', '0004_auto_20210204_1750'),
+    ]
+
+    operations = [
+    ]
```
Comment (ashwin1111) [inline apps/sage_intacct/migrations/0005_merge_20210204_1933.py:14]: same

File: apps/sage_intacct/tasks.py [hunk 6]
```diff
@@ -500,3 +510,416 @@ def create_charge_card_transaction(expense_group, task_log):
...
+            schedule: Schedule = Schedule.objects.filter(
+                func='apps.sage_intacct.tasks.create_sage_intacct_reimbursement',
+                args='{}'.format(workspace_id)
+            ).first()
+
+            if schedule:
+                schedule.delete()
+
+
+def get_all_sage_intacct_bill_ids(sage_objects):
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:776]: is it a list / dict / model, please to fix this in all functions

File: apps/sage_intacct/utils.py [hunk 7]
```diff
@@ -391,14 +414,28 @@ def post_bill(self, bill: Bill, bill_lineitems: List[BillLineitem]):
         created_bill = self.connection.bills.post(bill_payload)
         return created_bill
 
+    def get_bill(self, bill_id):
+        """
+        GET bill from SAGE
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:419]: intacct :P

File: apps/sage_intacct/utils.py [hunk 8]
```diff
@@ -391,14 +414,28 @@ def post_bill(self, bill: Bill, bill_lineitems: List[BillLineitem]):
         created_bill = self.connection.bills.post(bill_payload)
         return created_bill
 
+    def get_bill(self, bill_id):
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:417]: : str

File: apps/workspaces/migrations/0006_merge_20210204_1933.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('workspaces', '0005_workspacegeneralsettings_import_projects'),
+        ('workspaces', '0005_auto_20210204_1750'),
+    ]
+
+    operations = [
```
Comment (ashwin1111) [inline apps/workspaces/migrations/0006_merge_20210204_1933.py:13]: same

File: apps/fyle/migrations/0006_merge_20210204_1933.py (hunk 1 above)
Comment (Sravanksk) [reply apps/fyle/migrations/0006_merge_20210204_1933.py:14]: This migration file got created after resolving merge conflicts, this will just merge two migration files to one, so the `operations` field is empty - Will remove this migration file and will create a new one for my changes to avoid confusion

File: apps/fyle/serializers.py (hunk 2 above)
Comment (NileshPant1999) [reply apps/fyle/serializers.py]: no we are not using it anywhere removing it

**Review by ashwin1111 — APPROVED** [review]

**Review by Shwetabhk — APPROVED** [review]
